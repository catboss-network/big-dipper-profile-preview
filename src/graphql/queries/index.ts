import {
  parseQueryExample,
  QUERY_EXAMPLE,
  QUERY_EXAMPLE_TWO,
} from './query_example';

export const QUERIES = {
  QUERY_EXAMPLE,
  QUERY_EXAMPLE_TWO,
};

export const QUERY_PARSERS = {
  parseQueryExample,
};
