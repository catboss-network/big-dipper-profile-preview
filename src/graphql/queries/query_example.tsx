import * as R from 'ramda';
import { gql } from '@apollo/client';

export const QUERY_EXAMPLE = gql`
query GetCurrency {
  rates(currency: "USD") {
    currency
  }
}`;

export const QUERY_EXAMPLE_TWO = gql`
query GetRates {
  rates(currency: "CAD") {
    rate
  }
}`;

export const parseQueryExample = (data: any) => {
  const parsedData = R.pathOr([], ['rates'], data).slice(0, 1);
  return parsedData;
};
