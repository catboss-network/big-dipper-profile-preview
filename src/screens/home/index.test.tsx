import React from 'react';
import renderer from 'react-test-renderer';
import { ApolloProvider } from '@apollo/client';
import {
  createMockClient, MockApolloClient,
} from 'mock-apollo-client';
import { QUERIES } from '@graphql/queries';
import { wait } from '@tests/utils';
import Home from '.';

// ==================================
// global setup
// ==================================
let component;
let mockClient:MockApolloClient;

// ==================================
// mocks
// ==================================
jest.mock('@contexts', () => ({
  useSettingsContext: () => ({
  }),
}));

const mockExample = jest.fn().mockResolvedValue({
  data: {
    rates: [
      {
        currency: 'AKT',
      },
      {
        currency: 'UNI',
      },
    ],
  },
});

const mockExampleTwo = jest.fn().mockResolvedValue({
  data: {
    rates: [
      {
        rate: '0.1',
      },
      {
        rate: '0.2',
      },
    ],
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: Home', () => {
  beforeEach(async () => {
    mockClient = createMockClient();
    mockClient.setRequestHandler(
      QUERIES.QUERY_EXAMPLE_TWO,
      mockExampleTwo,
    );

    mockClient.setRequestHandler(
      QUERIES.QUERY_EXAMPLE,
      mockExample,
    );

    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <Home />
        </ApolloProvider>,
      );
    });
    await wait();
  });

  it('it renders', async () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(component.root.findAllByType('h1')[0].children).toEqual(['common:welcome']);
    expect(component.root.findAllByProps({
      className: 'currency',
    })[0].children).toEqual(['AKT']);
    expect(component.root.findAllByProps({
      className: 'rate',
    })[0].children).toEqual(['0.1']);
    expect(mockExample).toBeCalledTimes(1);
    expect(mockExampleTwo).toBeCalledTimes(1);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
