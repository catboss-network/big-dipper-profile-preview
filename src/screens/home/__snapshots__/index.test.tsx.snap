// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home it renders 1`] = `
<div
  className="makeStyles-root-4"
>
  <h1>
    common:welcome
  </h1>
  <h1
    className="MuiTypography-root MuiTypography-h1 MuiTypography-alignRight"
  >
    common:welcome
  </h1>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        change language
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        theme
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="currency"
  >
    AKT
  </div>
  <div
    className="rate"
  >
    0.1
  </div>
</div>
`;
